File: java_brain/src/main/java/org/netbrains/javabrain/neural/WorkingNeuron.java
package org.netbrains.javabrain.neural;

import org.netbrains.javabrain.activation.Activation;

public abstract class WorkingNeuron extends Neuron {
    protected final Activation activation;

    public WorkingNeuron(int layerIndex, int index, Activation activation) {
        super(layerIndex, index);
        this.activation = activation;
    }

    public void applyDelta() {
        for(Connection c : connections) {
            double adjustment = eta * delta * c.getInputNeuron().getValue();
            c.adjustWeight(adjustment);
        }
    }

    @Override
    public double activate() {
        double s = 0.0;
        for(Connection con : connections) {
            s += con.activate();
        }
        netInput = s;
        value = activation.activate(s);
        return value;
    }
}

File: swift_brains/swift_brains/WorkingNeuron.swift

class WorkingNeuron : Neuron {
    private let eta: Double = 1.0
    
    internal var netInput: Double = 0.0
    internal var connections: [Connection] = []
    internal var activation: Activation
    
    init(layerIndex: Int, index: Int, activation: Activation) {
        self.activation = activation
        super.init(layerIndex: layerIndex, index: index)
    }
    
    public func setConnections(connections: [Connection]) {
        self.connections = connections
    }
    
    public func getConnection(index: Int) -> Connection {
        return connections[index]
    }
    
    public func applyDelta() {
        for con in connections {
            let adjustment = eta * delta * con.getInputNeuron().getValue()
            con.adjustWeight(delta: adjustment)
        }
    }
    
    public override func activate() -> Double {
        var s = 0.0
        for con in connections {
            s += con.getInputNeuron().activate() * con.getWeight()
        }
        netInput = s
        value = activation.activate(x: s)
        return value
    }
}
